core/scss/components/modal/mixins.scss

```diff
@@ -105,13 +105,27 @@
   }
 }
 
-@mixin modal-color($bg: false, $border-color: false, $box-shadow: false) {
+@mixin modal-color(
+    $color: $md-white-100,
+    $color-css-var: --md-textColor-primary,
+    $bg: false,
+    $bg-css-var: --md-background-primary,
+    $border-color: $md-white-100,
+    $border-color-css-var: --md-separator-primary,
+    $box-shadow: false
+  ) {
+
+  color: $color;
+  color: var($color-css-var, $color);
+
   @if $bg {
     background-color: $bg;
+    background-color: var($bg-css-var, $bg);
   }
 
-  @if $border-color !=false {
+  @if $border-color-css-var !=false {
```

Comment:
I think this should check border-color and not the css var. I will leave it as it is for now, but it's worth double checking once you're back.